src/main/java/mate/academy/exception/CustomGlobalExceptionHandler.java

```diff
@@ -18,22 +18,29 @@
 public class CustomGlobalExceptionHandler {
     @ExceptionHandler(MethodArgumentNotValidException.class)
     protected ResponseEntity<Object> invalidArgument(MethodArgumentNotValidException ex) {
-        return buildResponseEntity(new BookApiException(BAD_REQUEST,
+        return buildResponseEntity(new BookApiErrorResponse(BAD_REQUEST,
                 LocalDateTime.now(), getErrorsMessage(ex.getBindingResult())));
     }
 
     @ExceptionHandler(EntityNotFoundException.class)
     protected ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException ex) {
-        return buildResponseEntity(new BookApiException(NOT_FOUND,
+        return buildResponseEntity(new BookApiErrorResponse(NOT_FOUND,
                 LocalDateTime.now(), ex.getMessage()));
     }
 
-    private ResponseEntity<Object> buildResponseEntity(BookApiException bookApiError) {
+    @ExceptionHandler(RegistrationException.class)
+    protected ResponseEntity<Object> handleRegistrationException(RegistrationException ex) {
```

Comment:
if I'm not mistaken, this exception is thrown when there's a duplicate email


in this case 404 is wrong and you should use 409 CONFLICT

Reply:
changed